### DISCUSSION TITLE
Add custom menu items to menu bar (macOS only?)

### DISCUSSION THREAD

**Discussion by ZontarOne** [kovidgoyal/kitty#6680] [Ideas]
Hi,
I'love the idea behind Kitty and I plan on using on my sysadmin work (not my main job, but it happens several times a day).
I am using a MacBookPro with MacOS 13.

 I've written some shell scripts that instruct (via kitty @ commands) kitty to split a window in 4, then do ssh to a different server in each one, and then launch a `tail` command on a specific log file.
Or another one where I instruct kitty to connect to several severs, cd to a specific dir, and launche the python virtualenv.

I'd like this script to be available (and launchable) from the Kitty menubar on macOS, let's say under the "shell" top menu (or another, custom menu) so I can launch them with my mouse, instead of having to launch the script from command line or using a shortcut (which I am sure to forget). This would allow me to share the script with the team (along with maybe an install script)  and everything would be self contained and easy to use.

Since I haven't found a viable way to do it, I propose that either an API is exposed to kittens, or a command is added to the remote controls of kitty.

Thans

**Comment by kovidgoyal** [comment]
dont see why this needs to be an API. Simply needs a configuration
 directive like map, except that the actions are mapped to menu bar
 entries. However, given this is macOS specific, I am not inclined to
 spend time on it as I prefer to focus on cross platform features.
 Patches welcome.

**Reply by ZontarOne** [reply]
Hi, I understand.
A configuration directive, anything will do. I understand that right now it would be macOS specific (even though I can see a niche use case also in Linux), so it's patch time.

I am not sure wether it would need to be implemented in the C part of the source code, or in the python part, a pointer would be welcome.
If it would be possible to write one in Python (maybe a kitten?) I could well try and do it. Can you just give me an indication wether it's feasible or not this way?

Thanks for the great software anyway

**Comment by kovidgoyal** [comment]
No, it would need Objective-C and python both.

**Comment by kovidgoyal** [comment]
Implemented in https://github.com/kovidgoyal/kitty/commit/f73d32e15d0eab1b061a6912af8eea0f69fdedfe

**Comment by ZontarOne** [comment]
thanks! will try it in from source code

**Comment by ZontarOne** [comment]
wonderful! very nice, it works as expected. thank you so much.
interesting looking at the code also

**Comment by mikesmithgh** [comment]
this is really cool!
